### ISSUE TITLE
Set common client configurations when initializing the connector.  

### ISSUE THREAD

**Issue by aneeshafedo** [ballerina-platform/openapi-tools#475]
**Description:**

Client initialization of the generated connectors only includes `auth` and `secureSocket` configurations as below. 
##### Current connector init function
```
    public type ClientConfig record {
        http:BearerTokenConfig|http:OAuth2RefreshTokenGrantConfig authConfig;
        http:ClientSecureSocket secureSocketConfig?;
    };
    public isolated function init(ClientConfig clientConfig, string serviceUrl = "https://api.zoom.us/v2") returns error? {
        http:ClientSecureSocket? secureSocketConfig = clientConfig?.secureSocketConfig;
        http:Client httpEp = check new (serviceUrl, { auth: clientConfig.authConfig, secureSocket: secureSocketConfig });
        self.clientEp = httpEp;
    }
```
Other than these configurations there are set of other configurations provided through Ballerina [http:ClientConfiguration](https://lib.ballerina.io/ballerina/http/1.1.0-beta.2/records/ClientConfiguration) record.  

Generated connector initialization should also facilitate these configurations.

##### Suggested connector init function. 
```
    public type ClientConfig record {|
        *http:ClientConfiguration;
        http:BearerTokenConfig|http:OAuth2RefreshTokenGrantConfig auth;
    |};
    public isolated function init(ClientConfig clientConfig, string serviceUrl = "https://api.zoom.us/v2") returns error? {
        http:Client httpEp = check new (serviceUrl, clientConfig);
        self.clientEp = httpEp;
    }
```
When creating a new instance of a connector with above suggested init function  while providing `auth` configurations as given below
```
http:BearerTokenConfig authConfig = {
        token: "<your-token>"
    };
    ClientConfig clientConfig = {auth: authConfig};
```
an error is thrown due to the issue mentioned [here](https://github.com/ballerina-platform/ballerina-lang/issues/32012).

```
                    {ballerina/lang.map}InherentTypeViolation {"message":"invalid value for record field 'auth': expected value of type '(http:BearerTokenConfig|http:OAuth2RefreshTokenGrantConfig)', found '()'"}

                        at ballerina.http.1_1_0-beta_2.CommonClientConfiguration:CommonClientConfiguration$gen$.<init>(http_types.bal:98)
                        aneesha.zoom.0_1_0.tests.test:testListMeetings(tests/test.bal:16)
```

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

**Comment by aneeshafedo** [comment]
Implementation is blocked until the issue mentioned [here](https://github.com/ballerina-platform/ballerina-lang/issues/32012) get fixed. 

This is generally affecting to all our connectors, both open-api and handwritten.

**Comment by aneeshafedo** [comment]
Until blocker from Ballerina language to implement the original solution is get fixed, OpenAPi connector initialization will be modified as below to support common client configurations

```
# Provides a set of configurations for controlling the behaviours when communicating with a remote HTTP endpoint.
public type ClientConfig record {|
    # Configurations related to client authentication
    http:BearerTokenConfig|http:OAuth2RefreshTokenGrantConfig auth;
    # The HTTP version understood by the client
    string httpVersion = "1.1";
    # Configurations related to HTTP/1.x protocol
    http:ClientHttp1Settings http1Settings = {};
    # Configurations related to HTTP/2 protocol
    http:ClientHttp2Settings http2Settings = {};
    # The maximum time to wait (in seconds) for a response before closing the connection
    decimal timeout = 60;
    # The choice of setting `forwarded`/`x-forwarded` header
    string forwarded = "disable";
    # Configurations associated with Redirection
    http:FollowRedirects? followRedirects = ();
    # Configurations associated with request pooling
    http:PoolConfiguration? poolConfig = ();
    # HTTP caching related configurations
    http:CacheConfig cache = {};
    # Specifies the way of handling compression (`accept-encoding`) header
    http:Compression compression = http:COMPRESSION_AUTO;
    # Configurations associated with the behaviour of the Circuit Breaker
    http:CircuitBreakerConfig? circuitBreaker = ();
    # Configurations associated with retrying
    http:RetryConfig? retryConfig = ();
    # Configurations associated with cookies
    http:CookieConfig? cookieConfig = ();
    # Configurations associated with inbound response size limits
    http:ResponseLimitConfigs responseLimits = {};
    #SSL/TLS-related options
    http:ClientSecureSocket? secureSocket = ();
|};

public isolated client class Client {
    final http:Client clientEp;
    # Gets invoked to initialize the `connector`.
    # The connector initialization requires setting the API credentials. 
    #
    # + clientConfig - The configurations to be used when initializing the `connector`
    # + serviceUrl - URL of the target service
    # + return - An error at the failure of client initialization
    public isolated function init(ClientConfig clientConfig, string serviceUrl = "https://api.zoom.us/v2") returns error? {
        http:Client httpEp = check new (serviceUrl, clientConfig);
        self.clientEp = httpEp;
    }
```

**Comment by lnash94** [comment]
This design https://github.com/ballerina-platform/openapi-connectors/pull/737/files was introduced to avoid the break by adding a new field to the parent record in the client configuration.
Therefore we can close this PR in the favour of this issue https://github.com/ballerina-platform/openapi-tools/issues/1088 and PR https://github.com/ballerina-platform/openapi-tools/pull/1095

This fix available with swanlake 2201.2.3 onwards
